File: do_replace_lab.js
import fs from 'fs';
let c = fs.readFileSync('src/views/LabView.tsx', 'utf-8');

c = c.replace(/<button onClick=\{async \(\) => \{\s*if\(true\)\s*\{\s*await localDB\.deleteFromCollection\('lab_records',[^}]+;\s*const updated = await localDB\.getCollection\('lab_records'\);\s*setRecords\(updated\);\s*\}\s*\}\}[^>]*>\s*<Trash2[^>]*>\s*<\/button>/g, `<RecordActions onDelete={async () => { await localDB.deleteFromCollection('lab_records', r.id); const updated = await localDB.getCollection('lab_records'); setRecords(updated); }} />`);

c = c.replace(/<button onClick=\{async \(\) => \{\s*if\(true\)\s*\{\s*await localDB\.deleteFromCollection\('lab_records', record\.id\);\s*const updated = await localDB\.getCollection\('lab_records'\);\s*setRecords\(updated\);\s*\}\s*\}\}[^>]*>\s*<Trash2[^>]*>\s*<\/button>/g, `<RecordActions onDelete={async () => { await localDB.deleteFromCollection('lab_records', record.id); const updated = await localDB.getCollection('lab_records'); setRecords(updated); }} />`);

fs.writeFileSync('src/views/LabView.tsx', c);

File: do_replace_lab.py
import re

with open('src/views/LabView.tsx', 'r', encoding='utf-8') as f:
    text = f.read()

def patch_component(comp_name, dep_list, filter_extract_regex, replace_tbody_regex):
    global text
    
    # 1. Inject pagination state
    state_injection = f"""
  const [currentPage, setCurrentPage] = useState<number>(1);
  const pageSize = 20;
  useEffect(() => {{ setCurrentPage(1); }}, {dep_list});
"""
    # Find the first useState in the component to inject after
    pattern_usestate = re.compile(r'(function ' + comp_name + r'[\s\S]*?const \[[^\]]+\] = useState[^;]*;)')
    
    def repl_usestate(m):
        return m.group(1) + state_injection
        
    text = pattern_usestate.sub(repl_usestate, text, count=1)
    
    # 2. Wrap the table rendering with an IIFE to capture filtered array
    # Wait, instead of IIFE, let's just find the `<tbody>...</tbody>` and `</table></div>` block
    # and replace the array mapping.
    
    # ... actually it's much easier to just do it manually for each if they are different.
